client/connection/manager.go

```diff
@@ -55,33 +68,72 @@ func (manager *connectionManager) Connect(consumerID, providerID identity.Identi
 		}
 	}()
 
-	proposal, err := manager.findProposalByProviderID(providerID)
+	cancelable := utils.NewCancelable()
+	manager.closeAction = utils.CallOnce(func() {
```

Comment:
`manager.closeAction` should be set before changing `manager.status = statusConnecting()` - that way we'll prevent a gap when we are in 'connecting' state but `closeAction` is not yet created, which will result in `warnOnClose`

Reply:
yes there is a tiny possibility. But mutexes and locking would be better. Lets leave it to other PR